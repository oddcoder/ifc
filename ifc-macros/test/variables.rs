use ifc_macros::ifc_block;

fn main() {
    ifc_block!{
        let _x : usize = 5;
        #[IFC(High)]
        let _y = 6;
        #[IFC(Low)]
        let _z = 7;
        let mut m : Vec<u8> = Vec::new();
    };
    // This is little hack that triggers compiler errors
    // This way we can detect compiler warnings that
    // should trigger
    compile_error!("Warning_Error");
}