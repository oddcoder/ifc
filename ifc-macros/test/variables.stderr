warning: All variables are `High` by default
 --> test/variables.rs:6:15
  |
6 |         #[IFC(High)]
  |               ^^^^
  |
  = help: Try removing `High`

error: Warning_Error
  --> test/variables.rs:15:5
   |
15 |     compile_error!("Warning_Error");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
